porch/pkg/git/git.go

```diff
@@ -296,15 +292,16 @@ func (r *gitRepository) CreatePackageRevision(ctx context.Context, obj *v1alpha1
 		return nil, fmt.Errorf("failed to create packagerevision: %w", err)
 	}
 
-	// TODO use git branches to leverage uniqueness
+	packagePath := filepath.Join(r.directory, obj.Spec.PackageName)
```

Comment:
I updated the PR in a way that hopefully makes it more obvious what is going on - the package name is now treated as relative to the directory registered with the repo. 

So the package path stored in our structs contains the base directory. And when we get the package name, we get it relative to the base directory.

Reply:
This seems to have triggered some test failures, I'm looking into them